app/gui/src/entrypoint.ts

```diff
@@ -1,57 +1,154 @@
-  window.clearTimeout(scamWarningHandle)
-  scamWarningHandle = window.setTimeout(printScamWarning, SCAM_WARNING_TIMEOUT)
-})
+  printScamWarning()
+  let scamWarningHandle = 0
 
-/** The entrypoint into the IDE. */
-function main() {
+  window.addEventListener('resize', () => {
+    window.clearTimeout(scamWarningHandle)
+    scamWarningHandle = window.setTimeout(printScamWarning, SCAM_WARNING_TIMEOUT)
+  })
+}
+
+function setupSentry() {
+  if (
+    !detect.IS_DEV_MODE &&
+    process.env.ENSO_CLOUD_SENTRY_DSN != null &&
+    process.env.ENSO_CLOUD_API_URL != null
+  ) {
+    const version: unknown = import.meta.env.ENSO_IDE_VERSION
+    sentry.init({
+      dsn: process.env.ENSO_CLOUD_SENTRY_DSN,
+      environment: process.env.ENSO_CLOUD_ENVIRONMENT,
+      release: version?.toString() ?? 'dev',
+      integrations: [
+        sentry.reactRouterV6BrowserTracingIntegration({
+          useEffect,
+          useLocation,
+          useNavigationType,
+          createRoutesFromChildren,
+          matchRoutes,
+        }),
+        sentry.extraErrorDataIntegration({ captureErrorCause: true }),
+        sentry.replayIntegration(),
+        new sentry.BrowserProfilingIntegration(),
+      ],
+      profilesSampleRate: SENTRY_SAMPLE_RATE,
+      tracesSampleRate: SENTRY_SAMPLE_RATE,
+      tracePropagationTargets: [process.env.ENSO_CLOUD_API_URL.split('//')[1] ?? ''],
+      replaysSessionSampleRate: SENTRY_SAMPLE_RATE,
+      replaysOnErrorSampleRate: 1.0,
+      beforeSend: (event) => {
+        if (
+          (event.breadcrumbs ?? []).some(
+            (breadcrumb) =>
+              breadcrumb.type === 'http' &&
+              breadcrumb.category === 'fetch' &&
+              breadcrumb.data &&
+              breadcrumb.data.status_code === HTTP_STATUS_BAD_REQUEST &&
+              typeof breadcrumb.data.url === 'string' &&
+              new URL(breadcrumb.data.url).host === API_HOST,
+          )
+        ) {
+          return null
+        }
+        return event
+      },
+    })
+  }
+}
+
+function configureAnimations() {
+  const areAnimationsDisabled =
+    window.DISABLE_ANIMATIONS === true ||
+    localStorage.getItem('disableAnimations') === 'true' ||
+    false
+
+  MotionGlobalConfig.skipAnimations = areAnimationsDisabled
+
+  if (areAnimationsDisabled) {
+    document.documentElement.classList.add('disable-animations')
+  } else {
+    document.documentElement.classList.remove('disable-animations')
+  }
+}
+
+function createQueryClientOfPersistCache() {
+  const store = idbKeyval.createStore('enso', 'query-persist-cache')
+  return createQueryClient({
+    persisterStorage: {
+      getItem: async (key) => idbKeyval.get(key, store),
+      setItem: async (key, value) => idbKeyval.set(key, value, store),
+      removeItem: async (key) => idbKeyval.del(key, store),
+      clear: async () => idbKeyval.clear(store),
+    },
+  })
+}
+
+function imNotSureButPerhapsFixingRefreshingWithAuthentication() {
```

Comment:
@somebody1234 I don't know what exactly the snippet below fixes. The Note is not clear to me.
I tracked that you are the original author, could you explain (and make a better name for this method :) )